src/scripts/main.js

```diff
@@ -1,3 +1,319 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+// #region sort
+
+const thead = document.querySelector('thead');
+const titles = thead.querySelectorAll('th');
+const nameTit = titles[0];
+const positionTit = titles[1];
+const officeTit = titles[2];
+const ageTit = titles[3];
+const salaryTit = titles[4];
+const sortState = {
+  names: 'desc',
+  position: 'desc',
+  office: 'desc',
+  age: 'desc',
+  salary: 'desc',
+};
+let employees = tbody.querySelectorAll('tr');
+
+function sortByAlphabet([...list], n, direction) {
+  list.sort((a, b) => {
+    const nameA = a.children[n].textContent;
+    const nameB = b.children[n].textContent;
+
+    return direction === 'asc'
+      ? nameA.localeCompare(nameB)
+      : nameB.localeCompare(nameA);
+  });
+
+  const parentList = list[0].parentNode;
+
+  list.forEach((item) => {
+    parentList.appendChild(item);
+  });
+}
+
+function sortByNumber([...list], n, direction) {
+  list.sort((a, b) => {
+    const numbA = Number(a.children[n].textContent.replace(/[^0-9.-]+/g, ''));
+    const numbB = Number(b.children[n].textContent.replace(/[^0-9.-]+/g, ''));
+
+    return direction === 'asc' ? numbA - numbB : numbB - numbA;
+  });
+
+  const parentList = list[0].parentNode;
+
+  list.forEach((item) => {
+    parentList.appendChild(item);
+  });
+}
+
+nameTit.addEventListener('click', () => {
+  const direction = sortState.names === 'asc' ? 'desc' : 'asc';
+
+  employees = tbody.querySelectorAll('tr');
+  sortByAlphabet(employees, 0, direction);
+  sortState.names = direction;
+});
+
+positionTit.addEventListener('click', () => {
+  const direction = sortState.position === 'asc' ? 'desc' : 'asc';
+
+  employees = tbody.querySelectorAll('tr');
+  sortByAlphabet(employees, 1, direction);
+  sortState.position = direction;
+});
+
+officeTit.addEventListener('click', () => {
+  const direction = sortState.office === 'asc' ? 'desc' : 'asc';
```

Comment:
When a new column header is clicked, the sort direction should be reset to 'asc'. This is part of the task requirements.